plugins/GSdx/GSTextureCache.cpp

```diff
@@ -2046,30 +2045,39 @@ void GSTextureCache::SourceMap::RemoveAt(Source* s)
 	delete s;
 }
 
-// Query the PaletteMap for a valid Palette, then assign both palette texture pointer and clut copy pointer to the Source object
-void GSTextureCache::AttachPaletteToSource(Source* s, uint16 pal)
+// Query the PaletteMap for a valid Palette, then both its reference, clut copy pointer and eventually palette texture pointer to the Source object
+void GSTextureCache::AttachPaletteToSource(Source* s, uint16 pal, bool need_gs_texture)
 {
-	std::shared_ptr<Palette> p = m_palette_map.LookupPalette(pal);
+	std::shared_ptr<Palette> p = m_palette_map.LookupPalette(pal, need_gs_texture);
 	s->m_palette_obj = p;
```

Comment:
I just noticed the shared pointer usage here is a bit inefficient.
```
{
	std::shared_ptr<Palette> p = m_palette_map.LookupPalette(pal, need_gs_texture);  // copy, +1 ref
	s->m_palette_obj = p; // copy, +1 ref
	// Rest of code in function
	// p destroyed, -1 ref
}
```
It could be
```
{
	s->m_palette_obj = m_palette_map.LookupPalette(pal, need_gs_texture);  // copy, +1 ref
	// Rest of code in function
}
```